core/register_core_types.cpp

```diff
@@ -100,7 +100,7 @@ extern void unregister_variant_methods();
 void register_core_types() {
 
 	//consistency check
-	ERR_FAIL_COND(sizeof(Callable) > 16);
+	static_assert(sizeof(Callable) <= 16, "sizeof(Callable) <= 16");
```

Comment:
We now use C++17 so you can drop the hardcoded error message. (I assume C++17's `static_assert` will be able to show a meaningful error by itself.)

Reply:
Done, captain :+1: